scope/src/doctor/runner.rs

```diff
@@ -603,61 +606,89 @@ mod tests {
 
         let exit_code = run_groups.execute().await?;
         assert!(exit_code.did_succeed);
-
+        assert_eq!(
+            BTreeSet::from_iter(run_groups.all_paths),
+            exit_code.succeeded_groups
+        );
+        assert_eq!(BTreeSet::new(), exit_code.failed_group);
+        assert_eq!(BTreeSet::new(), exit_code.skipped_group);
         Ok(())
     }
 
     #[tokio::test]
     async fn test_execute_dep_fails_wont_run_others() -> Result<()> {
         let group_actions = BTreeMap::from([
             make_group_action(
-                "group_1",
-                make_action_run(ActionRunStatus::CheckFailedFixSucceedVerifyFailed),
+                "fails",
+                make_action_runs(ActionRunStatus::CheckFailedFixSucceedVerifyFailed),
             ),
-            make_group_action("group_2", will_not_run()),
-            make_group_action("group_3", will_not_run()),
+            make_group_action("skipped_1", will_not_run()),
+            make_group_action("skipped_2", will_not_run()),
         ]);
 
         let run_groups = RunGroups {
             group_actions,
             all_paths: vec![
-                "group_1".to_string(),
-                "group_2".to_string(),
-                "group_3".to_string(),
+                "fails".to_string(),
+                "skipped_1".to_string(),
+                "skipped_2".to_string(),
             ],
         };
 
         let exit_code = run_groups.execute().await?;
         assert!(!exit_code.did_succeed);
-
+        assert_eq!(BTreeSet::new(), exit_code.succeeded_groups);
+        assert_eq!(
+            BTreeSet::from(["fails"].map(str::to_string)),
+            exit_code.failed_group
+        );
+        assert_eq!(
+            BTreeSet::from(["skipped_1", "skipped_2"].map(str::to_string)),
+            exit_code.skipped_group
+        );
         Ok(())
     }
 
     #[tokio::test]
     async fn test_execute_branch_fails_but_other_branch_continues() -> Result<()> {
         let group_actions = BTreeMap::from([
-            make_group_action("group_1", make_action_run(ActionRunStatus::CheckSucceeded)),
             make_group_action(
-                "group_2",
-                make_action_run(ActionRunStatus::CheckFailedFixSucceedVerifyFailed),
+                "succeeds_1",
+                make_action_runs(ActionRunStatus::CheckSucceeded),
+            ),
+            make_group_action(
+                "fails",
+                vec![make_action_run(
+                    ActionRunStatus::CheckFailedFixSucceedVerifyFailed,
+                    false,
+                )],
+            ),
+            make_group_action(
+                "succeeds_2",
+                make_action_runs(ActionRunStatus::CheckSucceeded),
             ),
-            make_group_action("group_3", will_not_run()),
```

Comment:
This is curious, why `group_3` and `group_4` have different expectations.  I think the fix is correct, as I don't see any hints or constraints for a relationship between 2 -> 3 that would make it not run.  It might represent a change in behavior that I'm not seeing so it would be good to get  @noizwaves 👀 here as well.

Reply:
This test didn't test what it said it did when I found it. 

It appears that it was intended to test that non-`required` actions didn't prevent actions following them from running.  `group_2` was marked as `required` when I _think_ it was supposed to be marked an `required: false`. I can't think of a way that `group_2` would fail, `group_3` would be skipped, and then `group_4` would run and succeed, so I made my best guess at the intention of the test.

Happy to add another if we think there is some setup where the four groups that were here make sense.

Reply:
Group 4 was not being run. I discovered it when I added the additional assertions. 

```console
---- doctor::runner::tests::test_execute_branch_fails_but_other_branch_continues stdout ----
thread 'doctor::runner::tests::test_execute_branch_fails_but_other_branch_continues' panicked at scope/src/doctor/runner.rs:664:9:
assertion `left == right` failed
  left: {"succeeds_1", "succeeds_2"}
 right: {"succeeds_1"}
```